### ISSUE TITLE
Bug: (v8) TypeScript version warning shows with TS 5.5

### ISSUE THREAD

**Issue by hildjj** [typescript-eslint/typescript-eslint#9420]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
{
  "name": "typescript-eslint-bug-repro",
  "version": "0.0.0",
  "devDependencies": {
    "eslint": "9.5.0",
    "typescript": "5.5.2",
    "typescript-eslint": "8.0.0-alpha.30"
  }
}
```

### ESLint Config

```javascript
import tseslint from "typescript-eslint";

const config = tseslint.config(...tseslint.configs.recommendedTypeChecked);
config[0].languageOptions.parserOptions = {
  project: true
};
export default config;
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

running eslint . does not show any warning on SUPPORTED TYPESCRIPT VERSIONS

### Actual Result

SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.5.0

### Additional Info

This is a request for a port forward of the change from #9399 / #9400 to the v8 branch.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | 8.0.0-alpha.30 |

**Comment by voxpelli** [comment]
As I posted in the other issue at the very moment this issue was posted:

I think this “simply” amounts to merging `main` into `v8`, as seems to have been the process for a while

**Comment by JoshuaKGoldberg** [comment]
We've since merged the latest `main` into `v8`. TypeScript 5.5 shouldn't give a warning anymore. Thanks for filing!
